### ISSUE TITLE
Making List.Embedded References Consistent

### ISSUE THREAD

**Issue by MarcoWel** [VictorAvelar/mollie-api-go#257]
**Is your feature request related to a problem? Please describe.**
For some List objects you define the Embedded field as slice of pointers, for some as slice of objects. Is there a reason for this?

E.g.:

```go
// Same for CapturesList, ChargebacksList, InvoicesList
type PaymentList struct {
	Count    int `json:"count,omitempty"`
	Embedded struct {
		Payments []Payment
	} `json:"_embedded,omitempty"`
	Links PaginationLinks `json:"_links,omitempty"`
}
```

```go
// Same for SettlementList, BalanceTransactionsList
type RefundList struct {
	Count    int `json:"count,omitempty"`
	Embedded struct {
		Refunds []*Refund
	} `json:"_embedded,omitempty"`
	Links PaginationLinks `json:"_links,omitempty"`
}
```

**Describe the solution you'd like**
The definition should be consistent.

**Comment by VictorAvelar** [comment]
I think what you are describing is completely accurate, I'll try to sweep the library and make it consistent.

**Comment by VictorAvelar** [comment]
Closed with #327 - All list operation return a slice of pointers to the resource they list.
